xbmc/addons/PVRClient.cpp

```diff
@@ -601,50 +601,50 @@ const std::string& CPVRClient::GetFriendlyName(void) const
   return m_strFriendlyName;
 }
 
-PVR_ERROR CPVRClient::GetDriveSpace(long long *iTotal, long long *iUsed)
+PVRError CPVRClient::GetDriveSpace(long long *iTotal, long long *iUsed)
 {
   if (!m_bReadyToUse)
-    return PVR_ERROR_SERVER_ERROR;
+    return PVRError_SERVER_ERROR;
 
-  try { return m_pStruct->GetDriveSpace(iTotal, iUsed); }
+  try { return TranslateError(m_pStruct->GetDriveSpace(iTotal, iUsed)); }
   catch (std::exception &e) { LogException(e, __FUNCTION__); }
 
   /* default to 0 on error */
   *iTotal = 0;
   *iUsed  = 0;
 
-  return PVR_ERROR_UNKNOWN;
+  return PVRError_UNKNOWN;
 }
 
-PVR_ERROR CPVRClient::StartChannelScan(void)
+PVRError CPVRClient::StartChannelScan(void)
 {
   if (!m_bReadyToUse)
-    return PVR_ERROR_SERVER_ERROR;
+    return PVRError_SERVER_ERROR;
 
   if (!m_addonCapabilities.bSupportsChannelScan)
-    return PVR_ERROR_NOT_IMPLEMENTED;
+    return PVRError_NOT_IMPLEMENTED;
 
-  try { return m_pStruct->OpenDialogChannelScan(); }
+  try { return TranslateError(m_pStruct->OpenDialogChannelScan()); }
   catch (std::exception &e) { LogException(e, __FUNCTION__); }
 
-  return PVR_ERROR_UNKNOWN;
+  return PVRError_UNKNOWN;
 }
 
-PVR_ERROR CPVRClient::OpenDialogChannelAdd(const CPVRChannelPtr &channel)
+PVRError CPVRClient::OpenDialogChannelAdd(const CPVRChannelPtr &channel)
 {
   if (!m_bReadyToUse)
-    return PVR_ERROR_SERVER_ERROR;
+    return PVRError_SERVER_ERROR;
 
   if (!m_addonCapabilities.bSupportsChannelSettings)
-    return PVR_ERROR_NOT_IMPLEMENTED;
+    return PVRError_NOT_IMPLEMENTED;
 
-  PVR_ERROR retVal(PVR_ERROR_UNKNOWN);
+  PVRError retVal(PVRError_UNKNOWN);
   try
   {
     PVR_CHANNEL addonChannel;
     WriteClientChannelInfo(channel, addonChannel);
 
-    retVal = m_pStruct->OpenDialogChannelAdd(addonChannel);
+    retVal = TranslateError(m_pStruct->OpenDialogChannelAdd(addonChannel));
```

Comment:
from a performance pov this wrapping is not acceptable. in 99.9% of all cases the result is success and caling TranslateError is just overhead